src/impact-dashboard/ChartArea.tsx

```diff
@@ -25,22 +36,74 @@ const LegendContainer = styled.div`
   flex: 0 0 auto;
 `;
 
-const ChartArea: React.FC = () => {
+function combinePopulations(data: CurveData, columnIndex: number) {
+  return zip(
+    getAllValues(getColView(data.incarcerated, columnIndex)),
+    getAllValues(getColView(data.staff, columnIndex)),
+  ).map(([incarcerated, staff]) => incarcerated + staff);
+}
+
+const ChartArea: React.FC<> = () => {
+  const modelData = useEpidemicModelState();
+
+  const projectionData = calculateCurves(modelData);
+
+  const curveData = {
+    exposed: combinePopulations(projectionData, seirIndex.exposed),
+    fatalities: combinePopulations(projectionData, seirIndex.fatalities),
+    hospitalized: combinePopulations(projectionData, seirIndex.hospitalized),
+    infectious: combinePopulations(projectionData, seirIndex.infectious),
+  };
+
   const markColors = {
     exposed: Colors.green,
     fatalities: Colors.black,
     hospitalized: Colors.lightBlue,
     hospitalBeds: Colors.red,
     infectious: Colors.red,
   };
+
+  const [groupStatus, setGroupStatus] = useState({
+    exposed: false,
+    fatalities: false,
+    hospitalized: false,
+    infectious: false,
+  });
+
+  const [displayData, setDisplayData] = useState({});
+
+  const toggleGroup = (groupName: keyof typeof groupStatus) => {
+    setGroupStatus({ ...groupStatus, [groupName]: !groupStatus[groupName] });
+  };
+
+  const updateDisplayData = () => {
+    let filteredGroupStatus = Object.keys(groupStatus).filter(
+      (groupName) => groupStatus[groupName],
+    );
+    setDisplayData(
+      filteredGroupStatus.reduce(
+        (data, key) => Object.assign(data, { [key]: curveData[key] }),
+        {},
+      ),
+    );
+  };
+
+  useEffect(() => {
+    updateDisplayData();
+  }, [groupStatus]);
```

Comment:
as we discussed earlier, it would be preferable for the data processing to remain inside `CurveChartContainer` for now. Why can't this logic run inside `CurveChartContainer` if you were to pass it `groupStatus` as a prop instead of the entire `curveData` structure?

Reply:
I think it's better for the parent component to hold the data, makes it a lot easier to share data between different children, instead of sibling communication, I put this data initially on it's own context. but as per our discussion, that was overkill. but you're are totally right it may seem like not needed at the moment but it will things a lot easier going forward.

Reply:
ok but, the curve data isn't being shared between siblings. the only thing that's being shared is the toggle status. The chart should own the curve data because it is specific to what the chart displays. 

There is an argument to be made that the full projection data should be shared to avoid doing those calculations twice — in fact I left some TODOs to that effect so we could think about coming back to clean that up later — but that's not what you're doing here. IMO this is just unnecessarily creating a tight coupling between the chart component and its parent.

Reply:
Alright, will update.